### ISSUE TITLE
Setting background image rearranges monitors layout

### ISSUE THREAD

**Issue by stratosgear** [swaywm/sway#3949]
I have an external monitor attached "above" my laptop screen as:

```
sway output '"Samsung' Electric Company SyncMaster 'H1AK500000"' pos 0 0 res 1680x1050 scale 1
sway output '"Panasonic' Industry Company 0x96A2 '0x00000000"' pos 0 1050 res 2560x1440 scale 1
```

This works fine, with my external monitor appearing on top of my laptop screen.

But my script to randomly change wallpapers, running as:

```
sway output '"Samsung' Electric Company SyncMaster 'H1AK500000"' bg /tmp/rand1.png fill
sway output '"Panasonic' Industry Company 0x96A2 '0x00000000"' bg /tmp/rand2.png fill
```

although correctly setting the wallpapers, it rearranges my monitors into a left to right orientation.

I was expecting the screen geometry to remain the same, and the wallpaper to just change.

My only workaround for now, is to query the state of the screens, and set again all required params (pos, scale etc)

**Comment by RedSoxFan** [comment]
Can you please post a debug log?

**Comment by stratosgear** [comment]
Yes, sorry.

https://gist.github.com/stratosgear/5aa14a029cb4516ef4158461281cf049

The behavior is as this (and these are the only actions I took while I had the `sway -d 2> ~/sway.log` running:

Run a shell script that:

```
 sway output '"Goldstar Company Ltd 20EN33 303NDMTMM283"' pos 1920 0 res 1600x900 scale 1
 sway output '"Dell Inc. DELL U2412M 0FFXD46R5Y6S"' pos 0 900 res 1920x1200 scale 1
 sway output '"Panasonic Industry Company 0x96A2 0x00000000"' pos 1920 900 res 2560x1440 scale 1
```
which results in correct monitor placement with proper mouse pointer movement across the monitors.

Then I run another shell script that just:

```
sway output eDP-1 bg /tmp/screenshot.png fit
```
to set a background pic on "Panasonic Industry Company 0x96A2 0x00000000" (eDP-1)

At that point my cursor was constrained in one monitor only, and my various desktops were still attached to other monitors.  Pressing mod+numbers I was able to switch to other desktops and my mouse would move to the center of the screen.  But mouse behavior and monitor geometry was incorrect.

I did a `swaymsg -t get_outputs` to check the state of the monitors and I got:

```
Output eDP-1 'Panasonic Industry Company 0x96A2 0x00000000'
  Current mode: 2560x1440 @ 59.999001 Hz
  Position: 3520,0
  Scale factor: 1.000000
  Transform: normal
  Workspace: 6:
  Available modes:
    2560x1440 @ 48.000000 Hz
    2560x1440 @ 59.999001 Hz

Output DP-7 'Goldstar Company Ltd 20EN33 303NDMTMM283'
  Current mode: 1600x900 @ 60.000000 Hz
  Position: 1920,0
  Scale factor: 1.000000
  Transform: normal
  Workspace: 4:
  Available modes:
    720x400 @ 70.082001 Hz
    640x480 @ 59.939999 Hz
    640x480 @ 75.000000 Hz
    800x600 @ 56.250000 Hz
    800x600 @ 60.317001 Hz
    800x600 @ 75.000000 Hz
    832x624 @ 74.551003 Hz
    1024x768 @ 60.004002 Hz
    1024x768 @ 75.028999 Hz
    1280x720 @ 60.000000 Hz
    1152x864 @ 75.000000 Hz
    1280x800 @ 59.810001 Hz
    1440x900 @ 59.887001 Hz
    1600x900 @ 60.000000 Hz

Output DP-6 'Dell Inc. DELL U2412M 0FFXD46R5Y6S' (focused)
  Current mode: 1920x1200 @ 59.950001 Hz
  Position: 0,900
  Scale factor: 1.000000
  Transform: normal
  Workspace: 2:
  Available modes:
    720x400 @ 70.082001 Hz
    640x480 @ 59.939999 Hz
    800x600 @ 60.317001 Hz
    1024x768 @ 60.004002 Hz
    1280x960 @ 60.000000 Hz
    1280x1024 @ 60.020000 Hz
    1680x1050 @ 59.953999 Hz
    1600x1200 @ 60.000000 Hz
    1920x1080 @ 60.000000 Hz
    1920x1200 @ 59.950001 Hz
```

which is incorrect (check the `pos`).

Then I quit the `sway -d 2> ~/sway.log` window and you have the log above.

Thanks

**Comment by RedSoxFan** [comment]
The attached debug log appears to be running on the Wayland backend and does not appear to be relevant.

If you are using the 1.0 release, I think this may be fixed by #3905.

**Comment by RedSoxFan** [comment]
Closing since this should be fixed by #3905. If you are able to replicate on git master, feel free to reopen with a new debug log
